docs/abi-spec.rst

```diff
@@ -17,13 +17,42 @@ We assume the Application Binary Interface (ABI) is strongly typed, known at com
 
 This specification does not address contracts whose interface is dynamic or otherwise known only at run-time. Should these cases become important they can be adequately handled as facilities built within the Ethereum ecosystem.
 
-Function Selector
-=================
+The ABI is constructed using JSON formatting for ease of use and ease of adoption amongst developers of various levels of expertise.
+
+.. index:: ! abi; contract description
+
+Contract Description
+====================
+
+There are 4 main types of contract fields that a contract's ABI can be made aware of:
+
+* Functions
+* Fallback
+* Constructor
+* Events
+
+Functions
+=========
+
+Functions are the meat and potatoes of an EVM smart contract. They denote a signature from the name of the function and its inputs and outputs. This signature is formed using a function selector algorithm.
```

Comment:
"meat and potatoes"?

Also i don't really understand what the second sentence means.